### PR TITLE
feat: credit card number input

### PR DESCRIPTION
**Pull request by hevele-moda**
cardinput

### CODE REVIEW

File: src/components/CreditCardNumberInput/CreditCardNumberInput.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    const { gaps, lengths } = cardType;
+    if (!gaps || !lengths) return DEFAULT_MASK;
+
+    const length = lengths[lengths.length - 1];
+    const digits = Array(length).fill('9');
+    return [...gaps]
+      .reverse()
+      .reduce(
+        (mask, gapIndex) => [...mask.slice(0, gapIndex), ' ', ...mask.slice(gapIndex)],
```
Comment (noahschechter) [inline src/components/CreditCardNumberInput/CreditCardNumberInput.tsx:40]: I wonder if a reduceRight here with some fancy indexing math would also work in order to get rid of the reverse

Comment (hevele-moda) [reply src/components/CreditCardNumberInput/CreditCardNumberInput.tsx:40]: Yeah actually `reduceRight` should work here, let me try.

**Comment by dzucconi** [comment]
Overkill for sure, but it'd be cool if design included [tabular numerals](https://www.fonts.com/content/learning/fontology/level-3/numbers/proportional-vs-tabular-figures) in the typeface so that the masking is monospaced with [font-variant-numeric](https://developer.mozilla.org/en-US/docs/Web/CSS/font-variant-numeric).

(Hi!)

**Comment by hevele-moda** [comment]
> Overkill for sure, but it'd be cool if design included [tabular numerals](https://www.fonts.com/content/learning/fontology/level-3/numbers/proportional-vs-tabular-figures) in the typeface so that the masking is monospaced with [font-variant-numeric](https://developer.mozilla.org/en-US/docs/Web/CSS/font-variant-numeric).
> 
> (Hi!)

Hi Damon, good to see you here 😄 I like that idea.

**Pull request merged**